### DISCUSSION TITLE
Pagination Metadata for Join Fields

### DISCUSSION THREAD

**Discussion by matthijs166** [payloadcms/payload#10142] [Feature Requests & Ideas]
The new features in 3.0 are amazing!
However, there’s one thing I’d love to see improved or clarified: pagination for join fields.

For example, with two collections: Authors and Books. When I fetch an Author item, I’d like to display their first 10 books with pagination options for the rest. While the API fetch includes the first 10 books and a boolean indicating whether there’s a next page, it lacks the rest of the metadata for correct pagination.

This limitation makes it challenging to implement efficient pagination without resorting to multiple fetches. For example, I currently have to:

Fetch the Author with their initial 10 books.
Perform an additional query to the Books collection using a where clause for the metadata for pagination.

**Questions:**
Am I missing something in the documentation about how to handle this better?
Is there a way to include pagination metadata for join fields in the query response?

**Feature Request:**
It would be fantastic if join fields supported full pagination metadata out of the box, similar to how it works for standalone collections. This would significantly improve usability and eliminate the need for multiple on load fetches in scenarios like mine.

**Comment by r1tsuu** [comment]
The main reason for not doing this is that the `count` operation  is can be very _expensive_ to do. If we add this (and I think we can), it should be completely optional via:
```ts
joins: { 
  pages: {
    count: true
  }
}
```

**Reply by matthijs166** [reply]
Makes sense that it would be more expensive. An optional option would be awesome here!

Now I got the overhead of a second request fetching the exact same data and the extra complexity of handling the data.
Would make life much better in the future :)

Is my current method the way to handle the join pagination at the moment? Or do I miss something else?
So far I can see and understand the payload admin panel is also fetching the collection itself?

**Comment by matthijs166** [comment]
Looks like this PR adds this feature
https://github.com/payloadcms/payload/pull/10998

Will check it out tomorrow!!
